crates/parser/src/lib.rs

```diff
@@ -0,0 +1,513 @@
+use pest::{iterators::Pair, Parser};
+use pest_derive::Parser;
+
+pub mod ast;
+mod errors;
+
+use ast::{Assertion, AssertionKind, Comparison, TestDefinition};
+
+use wasm_bindgen::prelude::*;
+
+#[cfg(test)]
+mod tests;
+
+// This include forces recompiling this source file if the grammar file changes.
+// Uncomment it when doing changes to the .pest file
+const _GRAMMAR: &str = include_str!("zombienet.pest");
+
+#[derive(Parser)]
+#[grammar = "zombienet.pest"]
+pub struct ZombieNetParser;
+
+fn parse_name(pair: Pair<Rule>) -> &str {
+    return pair.into_inner().next().unwrap().as_str();
+}
+
+fn parse_within(pair: Pair<Rule>) -> u32 {
+    return pair
+        .into_inner()
+        .next()
+        .unwrap()
+        .as_str()
+        .parse::<u32>()
+        .unwrap();
+}
+
+fn parse_para_id(pair: Pair<Rule>) -> u16 {
+    return pair
+        .into_inner()
+        .next()
+        .unwrap()
+        .as_str()
```

Comment:
Shouldn't calling `.into_inner().as_str()` be sufficient? Or maybe I am misunderstanding

Reply:
https://docs.rs/pest/latest/pest/iterators/struct.Pairs.html

Reply:
Yes, that correct. Thanks!